api/mutations/channel/utils.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+import createLoaders from '../../loaders/';
+const loaders = createLoaders();
```

Comment:
**IMPORTANT: Do not do this!!!**

Loaders are meant to be per-request, and they cache loaded data for their lifetime. If you do this, you'll get stale data after the first load always.

Reply:
Actually did this intentionally to keep permissions separate from results, based on us wanting to move to use loaders in mutations. This is already a problem in the branded login mutation resolvers where we have to remember to do `loaders.foo.clear(id)` . Honestly I'd prefer to just never use loaders in mutations, just because it's the highest likelihood of returning stale data.

So, I don't mind removing this, but I'm not opposed to the idea of permissions checks having their own caching via loaders. We can discuss further monday.

Reply:
We have to remove this. Loaders are not meant to be used that way and it might lead to security issues down the line.